app/src/main/java/com/github/se/polyfit/viewmodel/qrCode/barcodeViewModel.kt

```diff
@@ -55,9 +70,39 @@ class BarCodeCodeViewModel @Inject constructor() : ViewModel() {
 
     val list = _listId.value?.toMutableList() ?: mutableListOf()
     list.add(0, id)
-    _listId.value = list
+    _listId.postValue(list)
     Log.v("QrCodeViewModel", "new list: ${_listId.value}")
   }
+
```

Comment:
for `getIngredients()`

- Avoid launching new coroutines inside observeForever. Instead, consider using a single coroutine scope for the entire operation.
- Instead of creating a mutable list inside the coroutine, modify the LiveData directly if possible.
- Add error handling to manage potential exceptions from API calls.